li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -532,13 +581,17 @@ public void close() {
     close(CONSUMER_CLOSE_MAX_TIMEOUT_SECS, TimeUnit.SECONDS);
   }
 
-  void closeNoLock(long timeout, TimeUnit timeUnit) {
-    if (_closed) {
-      return;
-    }
+  @Override
+  synchronized public void close(Duration timeout) {
+    close(timeout.toMillis(), TimeUnit.MILLISECONDS);
```

Comment:
i think we shuld do it the other way around - in upstream kafka the Duration-accepting methods have the better behaviour - we should have them be the "real" impls and have the 2-arg ones (long + unit) redirect to them.

Reply:
Done. Updated LiKafkaConsumerImpl#close() which does the same thing as well.